spacy/tokens/doc.pyx

```diff
@@ -1612,18 +1612,18 @@ cdef class Doc:
             for token_attr in doc_json["underscore_token"]:
                 if not Token.has_extension(token_attr):
                     Token.set_extension(token_attr)
-                for token in doc_json["underscore_token"][token_attr]:
-                    token_start = token_by_char(self.c, self.length, token["token_start"])
-                    value = token["value"]
-                    self[token_start]._.set(token_attr, value)
+                for token_data in doc_json["underscore_token"][token_attr]:
+                    start = token_by_char(self.c, self.length, token_data["start"])
+                    value = token_data["value"]
+                    self[start]._.set(token_attr, value)
                 
         if doc_json.get("underscore_span", {}):
             for span_attr in doc_json["underscore_span"]:
```

Comment:
I realize this isn't in the current commit, but can't this be replaced with a line like this?

```
for span_attr in doc_json.get("underscore_span", {}):
```

Same for the doc one further up. 

The current structure seems weird unless we are concerned about the case where `underscore_span` has been set to an unusuable / non-iterable value, like `null`, which I don't think is possible with data we generate?

Reply:
`underscore_span` and `underscore_token` are only set when `underscore` is.